### PR TITLE
Reformulate breakup without while loop

### PR DESCRIPTION
**Pull request by edejong-caltech**

### CODE REVIEW

File: PySDM/backends/impl_numba/methods/collisions_methods.py [hunk 1]
```diff
@@ -50,6 +50,84 @@ def coalesce(  # pylint: disable=too-many-arguments
...
+    min_volume,
+    breakup_rate,
+    breakup_rate_deficit,
+    warn_overflows,
+    volume,
+):  # pylint: disable=too-many-branches
+    # TODO #874: update the arguments for the calling function
+
+    # 1. find the max gamma that can be supported in 1 time step, add to rate,
+    #    and add remainder to the deficit (limits: max_xi, max_volume, min_volume)
```
Comment (slayoo) [inline PySDM/backends/impl_numba/methods/collisions_methods.py:71]: max_xi and max_volume are mentioned here, but not featured in the code, let's turn comments here into function names and function arguments for clarity?

File: PySDM/dynamics/collisions/breakup_fragmentations/always_n.py [hunk 2]
```diff
@@ -6,19 +6,47 @@
...
 
     def __call__(self, output, u01, is_first_in_pair):
         output *= self.zeros
         output += self.N_vec
+        self.x_plus_y.sum(self.particulator.attributes["volume"], is_first_in_pair)
+        self.vmax.max(self.particulator.attributes["volume"], is_first_in_pair)
+        self.frag_size.sum(self.particulator.attributes["volume"], is_first_in_pair)
+        self.frag_size /= self.N
+        # TODO #874
+        # self.particulator.backend.fragmentation_limiters(
```
Comment (slayoo) [inline PySDM/dynamics/collisions/breakup_fragmentations/always_n.py:28]: is it intentional that this is commented out?

File: tests/unit_tests/dynamics/collisions/test_sdm_breakup.py [hunk 3]
```diff
@@ -373,6 +385,76 @@ def test_multiplicity_overflow(backend=CPU()):
...
+        np.testing.assert_equal(
+            np.sum(
+                particulator.attributes["n"].to_ndarray()
+                * particulator.attributes["volume"].to_ndarray()
+            ),
+            np.sum(np.array(params["n_init"]) * np.array(params["v_init"])),
+        )
+
+    @staticmethod
+    # @pytest.mark.xfail(strict=True)
```
Comment (slayoo) [inline tests/unit_tests/dynamics/collisions/test_sdm_breakup.py:401]: let's remove such commented-out decorators

File: PySDM/backends/impl_numba/methods/collisions_methods.py [hunk 4]
```diff
@@ -50,6 +50,84 @@ def coalesce(  # pylint: disable=too-many-arguments
...
+        attributes[a, k] *= factor_k
+        attributes[a, j] *= factor_j
+
+    if overflow_flag:
+        if warn_overflows:
+            warn("overflow", __file__)
+
+
+@numba.njit(**{**conf.JIT_FLAGS, **{"parallel": False}})
+def break_up_old(  # pylint: disable=too-many-arguments,unused-argument
```
[the remark this answers is not in the document]
Comment (edejong-caltech) [reply PySDM/backends/impl_numba/methods/collisions_methods.py]: Addressed in @4955bb3

**Comment by slayoo** [comment]
Thank you @edejong-caltech for the updates and the new tests!
Just one more comment: are the different default values of `min_volume` in `collision.py` intensional (0.0 vs. 2 nm**3)?

**Pull request merged**